peachjam/js/components/ListFacets.vue

```diff
@@ -344,25 +78,72 @@ export default {
     inputChecked (key, value) {
       return this.getUrlParamValue(key).includes(value.toString());
     },
-
     clearAll () {
       window.location.href = `${window.location.origin}${window.location.pathname}`;
     },
-
     clearFacet (key) {
       const queryString = window.location.search;
       const urlParams = new URLSearchParams(queryString);
       urlParams.delete(key);
-      window.location.href = `${window.location.origin}${window.location.pathname}?${urlParams.toString()}`;
+      window.location.href = `${window.location.origin}${
+        window.location.pathname
+      }?${urlParams.toString()}`;
     },
     submit () {
       this.loading = true;
-      // On submit page refreshes
-      // Prevention of empty query params
-      for (const input of this.$refs.form.querySelectorAll('input')) {
-        if (!input.value) input.setAttribute('disabled', '');
-      }
+
       this.$refs.form.submit();
+    },
+    loadFacets () {
+      const facetTitle = {
+        alphabet: 'Alphabetical',
+        authors: 'Regional Body',
+        courts: 'Court',
+        docTypes: 'Document type',
+        judges: 'Judges',
+        years: 'Year'
+      };
+      const formatOptions = (options) => {
+        return options.map((option) => ({
+          label: option,
+          value: option
+        }));
+      };
+
+      for (const key in this.$props) {
+        if (this.$props[key].length && key !== 'alphabet') {
```

Comment:
Instead of iterating of `this.$props`, I think it would be better to iterate over `facetTitles` since the keys `facetTitles` are the same as `this.$props`.  Iteration over `facetTitles` is more explicit because we're using it to access the facet props of the component `alphabet, authors, courts, docTypes, judges, years`. It handles the complication of if we were to add more props to `ListFacets` that are not facet related.

Reply:
Perhaps we can simple the if logic abit
```
for (const key in facetTitles) {
        if (!this.props[key].length) return;
        if (key === 'alphabet') {
          //  alphabet logic

        } else if (key === 'years') {
          // years logic
        } else {
          // other logic
        }
}
```